x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/remoteaccess/RemoteAccessHeadersForCcsRestIT.java

```diff
@@ -324,6 +850,90 @@ private void setupClusterSettings(
         }
     }
 
+    private void expectApiKeyActionsAndHeadersForCluster(
```

Comment:
Optional nit: I think we can avoid some of the redundancy between this method and `expectActionsAndHeadersForCluster` -- the only difference between them is the assertion on `actualRemoteAccessAuthentication`. The method is quite complex so could be worth trying to further reconcile the two.

Reply:
https://github.com/elastic/elasticsearch/pull/93414/commits/8e4a4c3d6290267b02e56f224a2f0f8fc1f60dc5